src/ok.rs

```diff
@@ -605,3 +671,209 @@ fn memlock_rlimit() -> TestResult {
+                name: "Comparing TCB values".to_string(),
+                stat: TestState::Fail,
+                mesg: format!("The TCB versions did NOT match").into()
+            }
+        }
+    }
+    else if test == "rmp init"{
+        if status.is_rmp_init == 1 {
+            TestResult{
+                name: "Checking if RMP init".to_string(),
+                stat: TestState::Pass,
+                mesg: format!("RMP is init").into()
+                }
+        } else{
+            TestResult{
+                name: "Checking if RMP init".to_string(),
+                stat: TestState::Fail,
+                mesg: format!("RMP is NOT initialized").into()
+            }
+        }
+    }
+    else if test == "snp init"{
+        if status.state == 1{
+            TestResult{
+                name: "Checking if SNP init".to_string(),
+                stat: TestState::Pass,
+                mesg: format!("SNP is init").into()
+                }
+        }
+        else{
+             TestResult{
+                name:"Checking if SNP init".to_string(),
+                stat: TestState::Fail,
+                mesg: format!("SNP is NOT init").into()
+             }
+        }
+    }
+    else {
+        unreachable!()
+    }
+}
+
+fn sev_ioctl(test : &str) -> TestResult {
+    let status = match sev_platform_status(){
+        Ok(stat) => stat,
+        Err(e) => return TestResult {
+            name: "Looking up platform status".to_string(),
+            stat: TestState::Fail,
+            mesg: format!("Error finding platform status: {}",e).into()
+        }
+    };
+    if test == "SEV init" {
+        if status.state == State::Working {
+            TestResult {
+                name: "Checking if SEV init".to_string(),
+                stat: TestState::Pass,
+                mesg: format!("SEV is init and currently running a guest").into()
+            }
+        } else if status.state == State::Initialized {
+            TestResult {
+                name: "Checking if SEV init".to_string(),
+                stat: TestState::Pass,
+                mesg: format!("SEV is init, but not currently running a guest").into()
+            }
+        } else {
+            TestResult {
+                name: "Checking if SEV init".to_string(),
+                stat: TestState::Fail,
+                mesg: format!("SEV is NOT init").into()
+            }
+        }
+    } else if test == "Firmware Version"{
+        if status.build.version == 0.into() {
+            TestResult {
+                name: "Checking Firmware version".to_string(),
+                stat: TestState::Fail,
+                mesg: format!("Invalid Firmware version").into()
+            }
+        }
+        else{
+            TestResult {
+                name: "Checking SEV firmware version".to_string(),
+                stat: TestState::Pass,
+                mesg: format!("Sev firmware version: {}", status.build.version).into()
+            }
+        }
+    } 
```

Comment:
One thing that I did fail to mention about this test:

Place it under SNP tests and check that the firmware is a minimum of  1.51, that's the minimum needed to run SEV-SNP!